docs/decisions/managing-service-dependencies.md

```diff
@@ -0,0 +1,115 @@
+
+The `make run` target must function when only the root folder of the project is cloned, e.g. when using `git clone --filter=blob:none --sparse ...` and should use the latest images from GitHub (i.e. not require a local build) by default. Note that the initiating project could override the image tag used from "latest" to "dev" to use a local container.
+
+The `make run` target must not require any specific software to be installed on the host, other than docker, and any operating system commands can assume a bash shell (Windows users are recommended to use the Git Bash shell to run make commands).
+
+If a project anticipates that it could be part of a circular reference then it should defend for this by setting a temporary environment variable, e.g.
+
+```make
+include .env
+
+# Load dependent projects
+deps:
+ifndef PROJECT1_RUNNING
+	@export PROJECT1_RUNNING=true; \
+	for dep in "project2" "project3" ; do \
+		if [ ! -d ../$$dep ]; then \
+			git clone --filter=blob:none --sparse \
+				https://github.com/openfoodfacts/$$dep.git ../$$dep; \
+		fi; \
+		cd ../$$dep && make -e run; \
+	done
+endif
+
+# Called from other projects to start this project
+run: deps
+	docker compose -p project1 up -d
+```
+
+Note that the make `ifndef` command and comments mustn't be indented.
+
+Other projects may set the COMPOSE_PROJECT_NAME environment variable so it is important to use the `-p` parameter in `docker compose` to ensure that each dependency is loaded into its own a consistent project name.
+
+If a developer also wants to work on a referenced service they can easily convert it into a full clone using `git sparse-checkout disable`. Note that the `blob:none` filter is still applied but this should not present too many complications.
+
+#### Testing
+
+In most cases tests should not rely on external dependencies and these should be mocked where appropriate. However, if an external dependency is essential for specific integration or end-to-end tests then the above described mechanism should _not_ be used and it is the originating project's responsibility to pull in the required dependencies and also take care of any further chained dependencies.
+
+This could be achieved by adding a specific docker compose configuration invoked during tests to pull in additional dependencies, or by using a suitable [testcontainers](https://testcontainers.com/) implementation.
+
+### Confirmation
+
+Compliance with this approach should become self-evident as project start to include dependencies from other projects.
+
+If a dependency is found which requires more than just docker compose and a bash shell to run, then a bug should be logged in that repo.
+
+Tools like GitPod [multi-repository working](https://www.gitpod.io/docs/configure/workspaces/multi-repo) support this approach.
```

Comment:
We may also want to review for other novel development environments, like GitHub codespaces